softwareComponents/atoms/include/atoms/patterns.hpp

```diff
@@ -293,8 +296,9 @@ class ValuePtr {
 private:
     std::unique_ptr< T > _ptr;
 public:
+    ValuePtr() = default;
     ValuePtr( std::unique_ptr< T > ptr ): _ptr( std::move( ptr ) ) {}
-    ValuePtr( const T& t ): _ptr( std::make_unique< T >( t ) ) {}
+    ValuePtr( const T& t ): _ptr( std::unique_ptr<T>( t.clone() ) ) {}
```

Comment:
Proč by `Visitable` mělo fungovat jenom clonable objekty?

Reply:
To není `Visitable`, ale `ValuePtr` - a to je celý princip použití `ValuePtr`. Ale jinak - když mám `const &` na abstraktní třídu, tak nemůžu použít kopírovací konstruktor.

Reply:
Honza je idiot.